### PR TITLE
Stat[u]s Command

### LINKED ISSUE #129: Status Command
I was looking through Flare code and I thought it would be a cool idea to have a Status command, to be able to view the bot's status. I'm sure this will inevitably be shown on the Web panel, but I thought I'd make a suggestion for such a command here.

For reference, here's the Flare command: https://github.com/FlareBot/FlareBot/blob/master/src/main/java/stream/flarebot/flarebot/commands/informational/StatusCommand.java

What do y'all think?

### PR DESCRIPTION
**Pull request by JamesHawkinss**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Added/Changed
Added Status Command

I have made this a draft PR because I don't know if y'all even want this command, and I need feedback on what else to add to it.

Closes #129

### CODE REVIEW

File: src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.cascadebot.cascadebot.messaging.MessagingObjects;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+
+public class StatusCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        EmbedBuilder builder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+
+        builder.setTitle(context.getGuild().getName());
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java]: The title and thumbnail should be for the bot, not the guild

File: src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        builder.setThumbnail(context.getGuild().getIconUrl());
+
+        builder.addField("Cascade Version", CascadeBot.getVersion().toString(), true);
+        builder.addField("Total Shards", String.valueOf(CascadeBot.INS.getShardManager().getShardsTotal()), true);
+        builder.addField("Online Shards", String.valueOf(CascadeBot.INS.getShardManager().getShardsRunning()), true);
+        builder.addField("Average Ping", String.valueOf(CascadeBot.INS.getShardManager().getAveragePing()), true);
+        builder.addField("Shard Status", CascadeBot.INS.getShardManager().getStatus(context.getChannel().getJDA().getShardInfo().getShardId()).toString().toLowerCase(), true);
+        builder.addField("Shard ID", String.valueOf(context.getChannel().getJDA().getShardInfo().getShardId()), true);
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java]: You need to add 1 to this, shard IDs are zero based

File: src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import net.dv8tion.jda.core.EmbedBuilder;
+import net.dv8tion.jda.core.entities.Member;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.commandmeta.CommandContext;
+import org.cascadebot.cascadebot.commandmeta.ICommandMain;
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.messaging.MessagingObjects;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+
+public class StatusCommand implements ICommandMain {
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/informational/StatusCommand.java]: This should really be a core command

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        context.getTypedMessaging().replyInfo(builder);
+    }
+
+    @Override
+    public String command() {
+        return "status";
+    }
+
+    @Override
+    public Module getModule() {
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java]: This doesn't need to be declared for a core command

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        return Module.CORE;
+    }
+
+    @Override
+    public String description() {
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java]: Fill out the description, makes it easier to languagify :P

**Comment by JamesHawkinss** [comment]
> You've accidentally included changes from your other PR, try and make a branch for next time

Oops, sorry

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        EmbedBuilder builder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+        JDA jda = CascadeBot.INS.getShardManager().getShardById(context.getChannel().getJDA().getShardInfo().getShardId());
+
+        builder.setTitle(context.getSelfUser().getName());
+        builder.setThumbnail(context.getSelfUser().getAvatarUrl());
+
+        builder.addField("Cascade Version", CascadeBot.getVersion().toString(), true);
+        builder.addField("JDA Version", JDAInfo.VERSION, true);
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java]: Add versions for Lavaplayer maybe? Idk if they have it

Comment (JamesHawkinss) [reply src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java]: I can't find anything that might give a Lavaplayer version, sozzles.

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import org.cascadebot.cascadebot.music.LavaPlayerAudioSendHandler;
+import org.cascadebot.cascadebot.music.MusicHandler;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+
+public class StatusCommand implements ICommandCore {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        EmbedBuilder builder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+        JDA jda = CascadeBot.INS.getShardManager().getShardById(context.getChannel().getJDA().getShardInfo().getShardId());
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatusCommand.java]: This isn't used any where

**Comment by JamesHawkinss** [comment]
@binaryoverload #128 has been merged.

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        OperatingSystemMXBean osBean = ManagementFactory.getPlatformMXBean(OperatingSystemMXBean.class);
+
+        builder.setTitle(context.getSelfUser().getName());
+        builder.setThumbnail(context.getSelfUser().getAvatarUrl());
+
+        builder.addField("Total Guilds", String.valueOf(CascadeBot.INS.getShardManager().getGuilds().size()), true);
+        builder.addField("Active Guilds", String.valueOf(CascadeBot.INS.getShardManager().getGuildCache().size()), true);
+        builder.addField("Active Voice Channels", String.valueOf(CascadeBot.INS.getShardManager().getVoiceChannels().size()), true);
+        builder.addField("RAM Usage", FileUtils.byteCountToDisplaySize(Runtime.getRuntime().totalMemory()), true);
+        builder.addField("CPU Load", String.valueOf(osBean.getSystemCpuLoad()), true);
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java]: getProcessCpuLoad() Maybe?

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    public void onCommand(Member sender, CommandContext context) {
+        EmbedBuilder builder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+        OperatingSystemMXBean osBean = ManagementFactory.getPlatformMXBean(OperatingSystemMXBean.class);
+
+        builder.setTitle(context.getSelfUser().getName());
+        builder.setThumbnail(context.getSelfUser().getAvatarUrl());
+
+        builder.addField("Total Guilds", String.valueOf(CascadeBot.INS.getShardManager().getGuilds().size()), true);
+        builder.addField("Active Guilds", String.valueOf(CascadeBot.INS.getShardManager().getGuildCache().size()), true);
+        builder.addField("Active Voice Channels", String.valueOf(CascadeBot.INS.getShardManager().getVoiceChannels().size()), true);
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java]: This needs to be filtered :P

Comment (JamesHawkinss) [reply src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java]: I dunno how to do this, could you help me?

**Comment by JamesHawkinss** [comment]
@binaryoverload @weeryan17 did all the stuff that cascade told me to

File: src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java [hunk 10]
```diff
@@ -9,8 +9,10 @@
 import org.cascadebot.cascadebot.messaging.MessagingObjects;
 
 import com.sun.management.OperatingSystemMXBean;
+import org.cascadebot.cascadebot.music.MusicHandler;
 
 import java.lang.management.ManagementFactory;
+import java.util.logging.Filter;
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/core/StatsCommand.java]: Remove this :P

**Pull request merged**